src/test/java/GoblintMessagesTest.java

```diff
@@ -163,5 +172,61 @@ public void testConvertMessagesFromJson() throws IOException {
     }
 
 
+
+
+    @Test
+    public void testConvertFunctionsFromJson() throws IOException {
+        List<GoblintFunctionsResult> goblintFunctionsResults = readGoblintResponseJsonFunc();
+
+        when(goblintService.functions()).thenReturn(CompletableFuture.completedFuture(goblintFunctionsResults));
+        when(gobPieConfiguration.showCfg()).thenReturn(false);
```

Comment:
The functions list is only read when this is set to `true`. See [GoblintAnalysis.java](goblintService.functions()) lines 171-178, the function will return before calling `goblintService.functions()` if it is set to `false`.